src/Poisson2DSparseMatrix.cpp

```diff
@@ -6,16 +6,191 @@
     Poisson2DSparseMatrix::Poisson2DSparseMatrix(DGContext2D& dg, MeshManager& mshManager) {
+        const index_vector_type& bcType = mshManager.get_BCType();
+        buildPoissonOperator(dg, mshManager, bcType);
+    }
+
+    void Poisson2DSparseMatrix::buildBcRhs(DGContext2D& dg, const MeshManager& mshManager, const real_matrix_type& ubc, const real_matrix_type& qbc, const index_vector_type& bcType) {
+        const index_type Np = dg.numLocalPoints();
+        const index_type K = dg.numElements();
+        const index_type Nfaces = dg.numFaces();
+        const index_type Nfp = dg.numFacePoints();
+
+        real_matrix_type bcRhs(Np, K);
+        bcRhs = 0.0;
+
+        const real_matrix_type& Dr = dg.Dr();
+        const real_matrix_type& Ds = dg.Ds();
+
+        const real_matrix_type& rx = dg.rx();
+        const real_matrix_type& ry = dg.ry();
+        const real_matrix_type& sx = dg.sx();
+        const real_matrix_type& sy = dg.sy();
+        const real_matrix_type& J  = dg.jacobian();
+        const real_matrix_type& Fscale = dg.fscale();
+
+        const index_vector_type& EToE = mshManager.get_EToE();
+        const index_vector_type& EToF = mshManager.get_EToF();
+
+        blitz::firstIndex ii;
+        blitz::secondIndex jj;
+        blitz::thirdIndex kk;
+
+        const index_type N = dg.order();
+
+        const index_vector_type& vmapM = dg.vmapM();
+        const index_vector_type& vmapP = dg.vmapP();
+
+        const index_matrix_type& Fmask = dg.fmask();
+
+        real_tensor3_type massEdge(Np, Np, Nfaces);
+        massEdge = 0.0;
+
+        Nodes1DProvisioner nodes1d(Nfp-1, 1, -1., 1.);
+
+        real_matrix_type V1D1(Nfp, Nfp), V1Dinv1(Nfp, Nfp);
+        real_matrix_type V1D2(Nfp, Nfp), V1Dinv2(Nfp, Nfp);
+        real_matrix_type V1D3(Nfp, Nfp), V1Dinv3(Nfp, Nfp);
+
+        real_matrix_type myTemp(Nfp, Nfp);
+
+        VandermondeBuilders vandermonde;
+
+        const real_vector_type& r = dg.r(), s = dg.s();
+        const real_matrix_type& nx = dg.nx(), ny  = dg.ny();
+
+        real_vector_type edge1(Nfp), edge2(Nfp), edge3(Nfp);
+        for (index_type i=0; i < Nfp; ++i)
+            edge1(i) = r(Fmask(i, 0));
+
+        vandermonde.computeVandermondeMatrix(edge1, V1D1, V1Dinv1);
+        myTemp = blitz::sum(V1Dinv1(kk,ii)*V1Dinv1(kk,jj), kk);
+
+        for (index_type i=0; i < Nfp; ++i) {
+            for (index_type j=0; j < Nfp; ++j) {
+                massEdge(Fmask(i, 0), Fmask(j, 0), 0) = myTemp(i, j);
+            }
+        }
+
+        for (index_type i=0; i < Nfp; ++i) {
+            edge2(i) = r(Fmask(i, 1));
+        }
+
+        vandermonde.computeVandermondeMatrix(edge2, V1D2, V1Dinv2);
+        myTemp = blitz::sum(V1Dinv2(kk,ii)*V1Dinv2(kk,jj), kk);
+        for (index_type i=0; i < Nfp; ++i) {
+            for (index_type j=0; j < Nfp; ++j) {
+                massEdge(Fmask(i, 1), Fmask(j, 1), 1) = myTemp(i, j);
+            }
+        }
+
+        for (index_type i=0; i < Nfp; ++i)
```

Comment:
Missing brackets around for-loop.

Reply:
Will fix the brackets for consistency. Thanks.